components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/cache/SessionContextCache.java

```diff
@@ -138,6 +150,17 @@ private SessionContextCacheEntry getSessionFromDB(SessionContextCacheKey key) {
 
         if (sessionContextDO != null) {
             cacheEntry = new SessionContextCacheEntry(sessionContextDO);
+            if (cacheEntry.getContext() == null && cacheEntry.getOptimizedSessionContext() != null) {
```

Comment:
I think if the `OptimizedSessionContext` is available, we have to refer to that, rather than considering the unavailability of the old context

Reply:
Fixed with [`c43c465`](https://github.com/wso2/carbon-identity-framework/pull/4446/commits/c43c465fb4b8edc33bf309a076d8e8226b5df9e1)